File: evals/tasks/amazon_add_to_cart.ts
import { EvalFunction } from "@/lib/v3/types/evals";

export const amazon_add_to_cart: EvalFunction = async ({
  logger,
  debugUrl,
  sessionUrl,
  v3,
}) => {
  try {
    const page = v3.context.pages()[0];
    await page.goto(
      "https://browserbase.github.io/stagehand-eval-sites/sites/amazon/",
    );

    await v3.act({
      instruction: "click the 'Add to Cart' button",
    });

    await v3.act({
      instruction: "click the 'Proceed to checkout' button",
    });

    const currentUrl = await page.url();
    const expectedUrl =
      "https://browserbase.github.io/stagehand-eval-sites/sites/amazon/sign-in.html";

    return {
      _success: currentUrl === expectedUrl,
      currentUrl,
      debugUrl,
      sessionUrl,
      logs: logger.getLogs(),
    };
  } catch (error) {
    return {
      _success: false,
      error: error,
      debugUrl,
      sessionUrl,
      logs: logger.getLogs(),
    };
  } finally {
    await v3.close();
  }
};

File: packages/sdk-ruby/evals/tasks/act/amazon_add_to_cart.rb
# frozen_string_literal: true

# Port of packages/evals/tasks/bench/act/amazon_add_to_cart.ts

Evals.define_task("amazon_add_to_cart") do |t|
  t.page.goto("https://browserbase.github.io/stagehand-eval-sites/sites/amazon/")

  t.stagehand.act("click the 'Add to Cart' button")

  t.stagehand.act("click the 'Proceed to checkout' button")

  current_url = t.page.url
  expected_url = "https://browserbase.github.io/stagehand-eval-sites/sites/amazon/sign-in.html"

  t.logger.log("currentUrl #{current_url}")
  t.logger.log("expectedUrl #{expected_url}")

  { _success: current_url == expected_url, currentUrl: current_url }
end
